### PR TITLE
[LETS-72] Fix -Wsign-compare

### PR DESCRIPTION
**Pull request by adrianbrab**
http://jira.cubrid.org/browse/LETS-72

Fix bad compares between signed and unsigned variables.

### CODE REVIEW

File: src/base/ddl_log.c [hunk 1]
```diff
@@ -917,7 +917,7 @@ logddl_write_end_for_csql_fileinput (const char *fmt, ...)
 {
   FILE *fp = NULL;
   char buf[DDL_LOG_BUFFER_SIZE] = { 0 };
-  int len = 0;
+  unsigned int len = 0;
```
Comment (ribram) [inline src/base/ddl_log.c:920]: len stores the result of snprintf and snprintf returns a size_t.
```suggestion
  size_t len = 0;
```
As a result of len being unsigned, conditions comparing `len < 0` should be removed too.

File: src/cci/cci_handle_mng.c [hunk 2]
```diff
@@ -1205,7 +1205,7 @@ hm_create_health_check_th (char useSSL)
   rv = pthread_attr_setdetachstate (&thread_attr, PTHREAD_CREATE_DETACHED);
   rv = pthread_attr_setscope (&thread_attr, PTHREAD_SCOPE_SYSTEM);
 #endif /* WINDOWS */
-  rv = pthread_create (&health_check_th, &thread_attr, hm_thread_health_checker, (void *) useSSL);
+  rv = pthread_create (&health_check_th, &thread_attr, hm_thread_health_checker, (void *) (size_t) useSSL);
```
Comment (ribram) [inline src/cci/cci_handle_mng.c:1208]: hm_thread_health_checker handling of argument may be modified for clarity (at cci_handle_mng.c:1476):
```c
  char useSSL = *((char *) (&arg));
```
to:
```c
  char useSSL = ((size_t) arg) != 0 ? USESSL : NON_USESSL;
```

File: src/parser/csql_grammar.y [hunk 3]
```diff
@@ -151,8 +151,6 @@ extern int yybuffer_pos;
 
 #define STACK_SIZE	128
 
-#define PT_IS_SERIAL(op)	((op) == PT_NEXT_VALUE || (op) == PT_CURRENT_VALUE)
```
Comment (ribram) [inline src/parser/csql_grammar.y:154]: This seems to be a fix of a different type of warning (http://jira.cubrid.org/browse/LETS-73)

File: src/storage/file_io.h [hunk 4]
```diff
@@ -335,7 +335,7 @@ struct fileio_backup_buffer
   int dtype;			/* Set to the type (dir, file, dev) */
   INT64 iosize;			/* Optimal I/O pagesize for backup device */
   INT64 count;			/* Number of current buffered bytes */
-  INT64 voltotalio;		/* Total number of bytes that have been either read or written (current volume) */
+  UINT64 voltotalio;		/* Total number of bytes that have been either read or written (current volume) */
   INT64 alltotalio;		/* total for all volumes */
```
Comment (ribram) [inline src/storage/file_io.h:339]: Changing only voltotalio type to UINT64, but leaving other similar fields are INT64 is hard to justify. Better to opt for one of next options:

  - change all four fields related to count/size as UINT64 and solve all signed/unsigned warnings that are caused by the change.
  - leave them as INT64 and fix signed/unsigned warnings related to voltotalio.

**Review by ribram — APPROVED** [review]

**Review by hornetmj — APPROVED** [review]

File: src/base/ddl_log.c [hunk 5]
```diff
@@ -781,7 +781,7 @@ logddl_write ()
 	}
 
       len = logddl_create_log_msg (buf);
-      if (len < 0 || fwrite (buf, sizeof (char), len, fp) != len)
+      if (len < 0 || fwrite (buf, sizeof (char), len, fp) != (unsigned) len)
```
[the remark this answers is not in the document]
Comment (adrianbrab) [reply src/base/ddl_log.c:784]: Changed to size_t

File: src/base/ddl_log.c [hunk 6]
```diff
@@ -888,7 +888,7 @@ logddl_write_tran_str (const char *fmt, ...)
 	  len = DDL_LOG_BUFFER_SIZE;
 	}
 
-      if (len < 0 || fwrite (msg, sizeof (char), len, fp) != len)
+      if (fwrite (msg, sizeof (char), len, fp) != len)
```
[the remark this answers is not in the document]
Comment (adrianbrab) [reply src/base/ddl_log.c:891]: len is of type size_t  and comparing an unsigned variable with 0 is generating an warning

File: src/cci/cci_handle_mng.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrianbrab) [reply src/cci/cci_handle_mng.c:1208]: The up-casting is necessary to avoid the ```warning: cast to 'void *' from smaller integer type 'char'```  warning

**Pull request merged**